File: cpp/392_Is_Subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
		int sp = 0, tp = 0;			        
		int slen = s.size(), tlen = t.size();

		if (slen > tlen) return false;

		while (sp < slen && tp < tlen) {
			// early break
			if (tlen - tp < slen - sp) return false;
			if (s[sp] == t[tp]) ++sp; 
			++tp;
		}

		return sp == slen;
    }
};

File: cpp/392_Is_Subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        sp, tp = 0, 0
        slen, tlen = len(s), len(t)

        if slen > tlen:
            return False

        while sp < slen and tp < tlen:
            if tlen - tp < slen - sp:
                return False

            if s[sp] == t[tp]:
                sp += 1

            tp += 1

        return sp == slen
